core/src/main/java/com/amazon/opendistroforelasticsearch/sql/expression/datetime/DateTimeFunction.java

```diff
@@ -114,28 +361,57 @@ private FunctionResolver timestamp() {
   }
 
   /**
-   * Specify a start date and add a temporal amount to the date.
-   * The return type depends on the date type and the interval unit. Detailed supported signatures:
-   * (DATE, DATETIME/TIMESTAMP, INTERVAL) -> DATETIME
-   * (DATE, LONG) -> DATE
-   * (DATETIME/TIMESTAMP, LONG) -> DATETIME
+   * TO_DAYS(STRING/DATE/DATETIME/TIMESTAMP). return the day number of the given date.
    */
-  private FunctionResolver adddate() {
-    return define(BuiltinFunctionName.ADDDATE.getName(),
-        impl(nullMissingHandling(DateTimeFunction::exprAddDateInterval), DATE, DATE, INTERVAL),
-        impl(nullMissingHandling(DateTimeFunction::exprAddDateInterval), DATETIME, DATE, INTERVAL),
-        impl(nullMissingHandling(DateTimeFunction::exprAddDateInterval),
-            DATETIME, DATETIME, INTERVAL),
-        impl(nullMissingHandling(DateTimeFunction::exprAddDateInterval),
-            DATETIME, TIMESTAMP, INTERVAL),
-        impl(nullMissingHandling(DateTimeFunction::exprAddDateDays), DATE, DATE, LONG),
-        impl(nullMissingHandling(DateTimeFunction::exprAddDateDays), DATETIME, DATETIME, LONG),
-        impl(nullMissingHandling(DateTimeFunction::exprAddDateDays), DATETIME, TIMESTAMP, LONG)
+  private FunctionResolver to_days() {
+    return define(BuiltinFunctionName.TO_DAYS.getName(),
+        impl(nullMissingHandling(DateTimeFunction::exprToDays), LONG, STRING),
+        impl(nullMissingHandling(DateTimeFunction::exprToDays), LONG, TIMESTAMP),
+        impl(nullMissingHandling(DateTimeFunction::exprToDays), LONG, DATE),
+        impl(nullMissingHandling(DateTimeFunction::exprToDays), LONG, DATETIME));
+  }
+
+  /**
+   * YEAR(STRING/DATE/DATETIME/TIMESTAMP). return the year for date (1000-9999).
+   */
+  private FunctionResolver year() {
+    return define(BuiltinFunctionName.YEAR.getName(),
+        impl(nullMissingHandling(DateTimeFunction::exprYear), INTEGER, DATE),
+        impl(nullMissingHandling(DateTimeFunction::exprYear), INTEGER, DATETIME),
+        impl(nullMissingHandling(DateTimeFunction::exprYear), INTEGER, TIMESTAMP),
+        impl(nullMissingHandling(DateTimeFunction::exprYear), INTEGER, STRING)
     );
   }
 
+  /**
+   * ADDDATE function implementation for ExprValue.
+   *
+   * @param date ExprValue of String/Date/Datetime/Timestamp type.
+   * @param expr ExprValue of Interval type, the temporal amount to add.
+   * @return Datetime resulted from expr added to date.
+   */
+  private ExprValue exprAddDateInterval(ExprValue date, ExprValue expr) {
+    ExprValue exprValue = new ExprDatetimeValue(date.datetimeValue().plus(expr.intervalValue()));
```

Comment:
adddate, date_add, date_sub & subdate need to return date/datetime according to calculated result. Here, I have returned date if time is zero for these functions to get correct result. But the return type will be still be datetime in schema because multiple return type can't be added in define function for the same input parameters.